server.js

```diff
@@ -14,10 +17,72 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.post("/messages", (req, res) => {
+  if(req.body.from.length == 0){
+    res.status(400).json("Please enter a from property")
+    return
+  }
+  if(req.body.text.length == 0){
+    res.status(400).json("please enter a text property")
+    return
+  }
+  let message = {
+    id: messages.length,
+    from: req.body.from,
+    text: req.body.text,
+  };
+  messages.unshift(message);
+  res.json("message has been send successfully");
```

Comment:
it is quite common to return the newly added message in the response. What you've done here is fine